custom_components/octopus_intelligent/__init__.py

```diff
@@ -67,3 +67,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     _LOGGER.debug("Octopus Intelligent System component setup finished")
     return True
 
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Called when the config entry is removed (the integration is deleted)."""
+    octopus_system: OctopusIntelligentSystem = (
+        hass.data[DOMAIN][entry.entry_id][OCTOPUS_SYSTEM]
+    )
+    try:
+        await octopus_system.async_remove_entry()
+    except Exception as ex:  # pylint: disable=broad-exception-caught
+        _LOGGER.error(ex)
```

Comment:
The `async_remove_entry()` function is called when the integration’s config entries are deleted — i.e., when the user deletes the integration from Home Assistant (Settings > Integrations > Octopus Intelligent Tariff > Three vertical dots > Delete). The code added here (following the call chain) ensures that when this happens, the persistent data is deleted as well (JSON file at `config/.storage/octopus_intelligent.A-1234ABCD`).

Importantly, I have tested to confirm that this does **not** apply to HACS _updates_ (Home Assistant Community Store), using the custom repository installation method documented in this integration’s README file. This is because a HACS update does not involve deleting config entries (the config flow does not seem to get involved at all), so `async_remove_entry()` is not called. This means that when users update the integration through HACS, the persistent data is **not** removed, which is good (the intended behaviour).

If the user tries to delete the integration through the HACS user interface, they are asked by HACS to fist delete the integration through the HASS UI (Settings > Integrations > ...) prior to deleting the custom repository from HACS.

So everything works as intended, whether the integration is installed / deleted with or without HACS.